neurokit2/ecg/ecg_delineate.py

```diff
@@ -129,20 +129,24 @@ def ecg_delineate(
     # Remove NaN in Peaks, Onsets, and Offsets
     waves_noNA = waves.copy()
     for feature in waves_noNA.keys():
-        waves_noNA[feature] = [int(x) for x in waves_noNA[feature] if ~np.isnan(x)]
+        waves_noNA[feature] = [int(x) for x in waves_noNA[feature] if ~np.isnan(x) and x > 0]
 
     instant_peaks = signal_formatpeaks(waves_noNA, desired_length=len(ecg_cleaned))
     signals = instant_peaks
 
+    waves_sanitized = {}
+    for feature, values in waves.items():
+        waves_sanitized[feature] = [x for x in values if x > 0]
```

Comment:
@zen-juen this line removes all nan values too and thus causing the tests to fail. Could you also return x if x is np.nan

Reply:
I fixed it in the commit below